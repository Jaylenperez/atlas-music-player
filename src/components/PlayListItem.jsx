export default function PlayListItem({ title, artist, length }) {
  return (
    <div className="flex items-center justify-between rounded-md border-b border-gray-200 px-4 py-2 hover:bg-gray-50">
      <div>
        <h3 className="text-lg font-medium text-sky-400">{title}</h3>
        <p className="text-sm text-gray-500">{artist}</p>
      </div>
      <span className="text-sm text-gray-400">{length}</span>
    </div>
  );
}
