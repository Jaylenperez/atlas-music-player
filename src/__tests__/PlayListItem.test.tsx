// src/__tests__/PlayListItem.test.tsx
import React from "react";
import { render, screen } from "@testing-library/react";
import { describe, test, expect } from "vitest";
import PlayListItem from "../components/PlayListItem";

describe("PlayListItem component", () => {
  const sample = {
    title: "Starlight",
    artist: "Nova Echo",
    length: "4:20",
  };

  test("matches snapshot", () => {
    const { container } = render(
      <PlayListItem
        title={sample.title}
        artist={sample.artist}
        length={sample.length}
      />
    );
    expect(container).toMatchSnapshot();
  });

  test("renders title, artist, and length text", () => {
    render(
      <PlayListItem
        title={sample.title}
        artist={sample.artist}
        length={sample.length}
      />
    );
    expect(screen.getByText(sample.title)).toBeInTheDocument();
    expect(screen.getByText(sample.artist)).toBeInTheDocument();
    expect(screen.getByText(sample.length)).toBeInTheDocument();
  });

  test("applies correct Tailwind classes", () => {
    render(
      <PlayListItem
        title={sample.title}
        artist={sample.artist}
        length={sample.length}
      />
    );
    // Title should be an <h3> with text-sky-400
    const titleEl = screen.getByRole("heading", { level: 3, name: sample.title });
    expect(titleEl).toHaveClass("text-sky-400");

    // Artist should be a <p> with text-fuchsia-500
    const artistEl = screen.getByText(sample.artist);
    expect(artistEl.tagName.toLowerCase()).toBe("p");
    expect(artistEl).toHaveClass("text-fuchsia-500");

    // Length should be a <span> with text-fuchsia-300
    const lengthEl = screen.getByText(sample.length);
    expect(lengthEl.tagName.toLowerCase()).toBe("span");
    expect(lengthEl).toHaveClass("text-fuchsia-300");
  });
});
