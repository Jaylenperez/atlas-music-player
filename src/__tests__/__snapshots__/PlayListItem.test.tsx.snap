// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PlayListItem component > matches snapshot 1`] = `
<div>
  <div
    class="flex items-center justify-between rounded-md border-b border-gray-900 px-4 py-2 hover:bg-indigo-900"
  >
    <div>
      <h3
        class="text-lg font-medium text-sky-400"
      >
        Starlight
      </h3>
      <p
        class="text-sm text-fuchsia-500"
      >
        Nova Echo
      </p>
    </div>
    <span
      class="text-sm text-fuchsia-300"
    >
      4:20
    </span>
  </div>
</div>
`;
